answerking_app/views/category_views.py

```diff
@@ -16,33 +17,31 @@
 
 class CategoryListView(View):
     @csrf_exempt
-    def get(self, request: HttpRequest, *args, **kwargs) -> JsonResponse | HttpResponse:
+    def get(self, request: Request, *args, **kwargs) -> Response:
         categories: QuerySet[Category] = category_service.get_all()
         response: list[ReturnDict] = []
 
         if categories:
             response = [CategorySerializer(cat).data for cat in categories]
 
-        return JsonResponse(response, encoder=DjangoJSONEncoder, safe=False)
+        return Response(response, status=status.HTTP_200_OK)
 
     @csrf_exempt
-    def post(
-        self, request: HttpRequest, *args, **kwargs
-    ) -> JsonResponse | HttpResponse:
+    def post(self, request: Request, *args, **kwargs) -> Response:
         try:
             body: dict | None = json.loads(request.body)
             if not body:
-                return HttpResponse(status=400)
+                return Response(status=status.HTTP_400_BAD_REQUEST)
         except JSONDecodeError as e:
```

Comment:
I might be wrong here, but I believe there are decorators and functions to handle JSON parsing in DRF

Reply:
found this link https://www.django-rest-framework.org/api-guide/parsers/#jsonparser

Reply:
Done